crates/sparrow-backend/src/logical_to_physical.rs

```diff
@@ -216,11 +248,32 @@ impl LogicalToPhysical {
                     )));
                 };
 
-                let args = node
-                    .args
-                    .iter()
-                    .map(|arg| self.visit(arg))
-                    .collect::<Result<_, _>>()?;
+                let args = if instruction != "record" {
```

Comment:
Records are annoying. I'd like to clean this up, but I'm not sure yet on the best way.

1. A record in the logical expression has `(record name1 value1 name2 value2 ...)` because Python doesn't distinguish between "literal arguments" and "arguments".
2. A record in the physical expression has `(record[name1, name2, ...] value1 value2 ...)`.

My current thinking is that I might add a trait in the `sparrow-interfaces` crate that is basically "how does the expression typecheck (logically and physically) and how does it convert", with a default implementation for things not registered. Not sure yet.